pwiz_tools/Skyline/Model/Results/ChromatogramCache.cs

```diff
@@ -1230,6 +1231,104 @@ public IEnumerable<ChromKeyIndices> GetChromKeys(MsDataFileUri msDataFilePath)
             }
         }
 
+        private ChromatogramCache UpdateOptimizationSteps(SrmDocument doc)
+        {
+            if (doc?.MeasuredResults?.Chromatograms == null || Version >= CacheFormatVersion.Seventeen)
+                return this;
+
+            // Get the set of MsDataFileUris belonging to ChromatogramSets with optimization functions.
+            var optFiles = new HashSet<MsDataFileUri>();

```

Comment:
I would probably make an array of booleans so that you can quickly check whether ChromGroupHeaderInfo.FileIndex was optimized or not, instead of having to look a MsDataFileUri up in a HashSet.
There probably won't be enough ChromGroupHeaderInfo's for that to make a difference, though.

Reply:
Sure, I'll change that.